config/rbac/aggregate_to_edit.yaml

```
@@ -0,0 +1,18 @@
+# Automatically aggregated to clusterRoles: edit -> admin
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+  name: hnc-edit
+rules:
+- apiGroups:
+  - hnc.x-k8s.io
+  resources:
+  - hierarchicalresourcequotas
+  - subnamespaceanchors
+  verbs:
+  - create
+  - update
+  - patch
+  - delete
```

Should this be '*'? I'm not 100% confident this is the full list :)

---

you can find the verbs with `kubectl api-resources -o wide`:
```console
% kubectl api-resources -o wide --namespaced --api-group=hnc.x-k8s.io
NAME                         SHORTNAMES   APIVERSION              NAMESPACED   KIND                        VERBS                                                        CATEGORIES
hierarchicalresourcequotas   hrq          hnc.x-k8s.io/v1alpha2   true         HierarchicalResourceQuota   delete,deletecollection,get,list,patch,create,update,watch   
hierarchyconfigurations                   hnc.x-k8s.io/v1alpha2   true         HierarchyConfiguration      delete,deletecollection,get,list,patch,create,update,watch   
subnamespaceanchors          subns        hnc.x-k8s.io/v1alpha2   true         SubnamespaceAnchor          delete,deletecollection,get,list,patch,create,update,watch   
```

so
```suggestion
  - create
  - update
  - patch
  - delete
  - deletecollection
```

---

yes, I wasnt sure if in edit I should put only the edit permissions and relay on the aggregation from view, or should I put *.  
I will change it to '*' for the simplicity

---

k8s [system:aggregate-to-edit](https://github.com/kubernetes/kubernetes/blob/0241da314e0e69817d66313b45a69c19d1ce7327/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L651) uses explicit verbs:  with [Write](https://github.com/kubernetes/kubernetes/blob/0241da314e0e69817d66313b45a69c19d1ce7327/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L35)

---

Yes so I wasn't sure if to put Write + relay on view aggregation, or ReadWrite.  
I think ReadWrite is better because then this ClusterRole can be used for other purposes and not only aggregation to edit.  
And ReadWrite is `'*'` anyways